sklearn/linear_model/tests/test_omp.py

```diff
@@ -85,14 +84,23 @@ def test_unreachable_accuracy():
 
 
 def test_bad_input():
-    assert_raises(ValueError, orthogonal_mp, X, y, tol=-1)
-    assert_raises(ValueError, orthogonal_mp, X, y, n_nonzero_coefs=-1)
-    assert_raises(ValueError, orthogonal_mp, X, y,
-                  n_nonzero_coefs=n_features + 1)
-    assert_raises(ValueError, orthogonal_mp_gram, G, Xy, tol=-1)
-    assert_raises(ValueError, orthogonal_mp_gram, G, Xy, n_nonzero_coefs=-1)
-    assert_raises(ValueError, orthogonal_mp_gram, G, Xy,
-                  n_nonzero_coefs=n_features + 1)
+    with pytest.error(ValueError):
+        orthogonal_mp(X, y, tol=-1)
+
+    with pytest.error(ValueError):
+        orthogonal_mp(X, y, n_nonzero_coefs=-1)
+
+    with pytest.error(ValueError):
+        orthogonal_mp(X, y, n_nonzero_coefs=n_features+1)
+
+    with pytest.error(ValueError):
+        orthogonal_mp(G, Xy, tol=-1)
+
+    with pytest.error(ValueError):
+        orthogonal_mp(G, Xy, n_nonzero_coefs=-1)
+
+    with pytest.error(ValueError):
+        orthogonal_mp(G, Xy, n_nonzero_coefs=n_features+1)
```

Comment:
Let's correct using `pytest.raises` and improve this test using pytest parametrization:

```python
@pytest.mark.parametrize("positional_params", [(X, y), (G, Xy)])
@pytest.mark.parametrize(
    "keyword_params",
    [{"tol": -1}, {"n_nonzero_coefs": -1}, {"n_nonzero_coefs": n_features + 1}]
)
def test_bad_input(positional_params, keyword_params):
    with pytest.raises(ValueError):
        orthogonal_mp(*positional_params, **keyword_params)
```